Description: Upstream changes introduced in version 0.1-4~wip
 This patch has been created by dpkg-source during the package build.
 Here's the last changelog entry, hopefully it gives details on why
 those changes were made:
 .
 libsml (0.1-4~wip) unstable; urgency=low
 .
   * Fixed bug in sml_message_body_free
 .
 The person named in the Author field signed this changelog entry.
Author: Steffen Vogel <info@steffenvogel.de>

---
The information above should follow the Patch Tagging Guidelines, please
checkout http://dep.debian.net/deps/dep3/ to learn about the format. Here
are templates for supplementary fields that you might want to add:

Origin: <vendor|upstream|other>, <url of original patch>
Bug: <url in upstream bugtracker>
Bug-Debian: http://bugs.debian.org/<bugnumber>
Bug-Ubuntu: https://launchpad.net/bugs/<bugnumber>
Forwarded: <no|not-needed|url proving that it has been forwarded>
Reviewed-By: <name and email of someone who approved the patch>
Last-Update: <YYYY-MM-DD>

--- libsml-0.1.orig/sml/src/sml_message.c
+++ libsml-0.1/sml/src/sml_message.c
@@ -166,7 +166,7 @@ sml_message_body *sml_message_body_init(
 
 void sml_message_body_write(sml_message_body *message_body, sml_buffer *buf) {
     sml_buf_set_type_and_length(buf, SML_TYPE_LIST, 2);
-	sml_u16_write(message_body->tag, buf);
+    sml_u16_write(message_body->tag, buf);
     
     switch (*(message_body->tag)) {
         case SML_MESSAGE_OPEN_REQUEST:
@@ -195,8 +195,6 @@ void sml_message_body_write(sml_message_
 
 void sml_message_body_free(sml_message_body *message_body) {
     if (message_body) {
-		sml_number_free(message_body->tag);
-		
         switch (*(message_body->tag)) {
             case SML_MESSAGE_OPEN_REQUEST:
                 sml_open_request_free((sml_open_request *) message_body->data);
@@ -224,6 +222,8 @@ void sml_message_body_free(sml_message_b
                 break;
         }
         
+        sml_number_free(message_body->tag);
+        
         free(message_body);
     }
 }
